cmd/spire-agent/cli/run/run.go

```diff
@@ -53,6 +53,10 @@ type agentConfig struct {
 
 	ConfigPath string
 	Umask      string `hcl:"umask"`
+
+	ProfilingEnabled string `hcl:"profiling_enabled"`
+	ProfilingPort    string `hcl:"profiling_port"`
+	ProfilingFreq    string `hcl:"profiling_freq"`
```

Comment:
I'm pretty sure HCL can support int and bool types, which should greatly simplify parsing and error handling code below.

Reply:
Yeah, I was pretty sure too, in fact, that was my first approach. But...there is a problem with the configuration merging logic. First you have: https://github.com/spiffe/spire/blob/c1e30d02ce3b76a80d1eae72bb64e0b7aaf6172b/cmd/spire-agent/cli/run/run.go#L165-L173
And then:
https://github.com/spiffe/spire/blob/c1e30d02ce3b76a80d1eae72bb64e0b7aaf6172b/cmd/spire-agent/cli/run/run.go#L175-L250
When I used int and a bool types, the code that overrides the fileConfig with the cliConfig, was overriding my values despite I wasn't setting any cliConfig value. That happened because if I set the int and bool values on fileConfig, when the code tries to merge cliConfig it had no way to determine if the cliConfig values (zero and false in this case)  were set by the user or were the golang default values, and then always ended up overriding my values (the ones set by fileConfig).